lib/incoming_form.js

```diff
@@ -537,7 +538,7 @@ IncomingForm.prototype._initJSONencoded = function() {
 
 IncomingForm.prototype._uploadPath = function(filename) {
   var buf = crypto.randomBytes(16);
-  var name = 'upload_' + buf.toString('hex');
+  var name = this.keepFilenames ? filename : 'upload_' + buf.toString('hex');
```

Comment:
Merging this patch would expose users of this flag to giving write access to their whole file system. 🤕 

You can try for yourself:

```
curl  -F "upload=@$(pwd)/harmless.jpg;filename=../hahah.exploit" 'localhost:8080/upload'
```

Please don't merge this or similar patches. If somebody wants to use the original filenames, they can implement it themselves at their own risk.